src/base/static/js/views/geometry-editor-view.js

```diff
@@ -0,0 +1,583 @@
+        coordinates.push([latLngs[0].lng, latLngs[0].lat]);
+        self.geometry = {
+          "type": "Polygon",
+          "coordinates": [coordinates]
+        }
+      } else if (self.geometryType === "polyline" ||
+          self.geometryType === "LineString") {
+        
+        var coordinates = buildCoords(layer);
+        self.geometry = {
+          "type": "LineString",
+          "coordinates": coordinates
+        }
+      } else if (self.geometryType === "marker" ||
+          self.geometryType === "Point") {
+
+        self.geometry = {
+          "type": "Point",
+          "coordinates": [layer._latlng.lng, layer._latlng.lat]
+        }
+      }
+    }
+
+    // ========== Drawing events ==========
+    this.map.on("draw:drawvertex", function(evt) {
+      self.numVertices++;
+
+      if (self.layerType === "polygon" && self.numVertices <= 2) {
+        self.displayHelpMessage("draw-polygon-continue-msg");
+      } else if (self.layerType === "polygon" && self.numVertices > 2) {
+        self.displayHelpMessage("draw-polygon-finish-msg");
+      } else if (self.layerType === "polyline" && self.numVertices === 1) {
+        self.displayHelpMessage("draw-polyline-continue-msg");
+      } else if (self.layerType === "polyline" && self.numVertices > 1) {
+        self.displayHelpMessage("draw-polyline-finish-msg");
+      }
+    });
+
+    this.map.on("draw:created", function(evt) {
+      self.resetWorkingGeometry();
+      self.setColorpicker();
+      self.hideIconToolbar();
+
+      self.geometryType = evt.layerType;
+      generateGeometry(evt.layer);
+      self.editingLayerGroup.addLayer(evt.layer);
+      self.swapToolbarVisibility();
+      self.displayHelpMessage("modify-geometry-msg");
+    });
+
+    this.map.on("draw:editvertex", function(evt) {
+      if (self.placeDetailView) {
+        self.placeDetailView.onModified();
+      }
+    });
+
+    this.map.on("draw:editmove", function(evt) {
+      if (self.placeDetailView) {
+        self.placeDetailView.onModified();
+      }
+    });
+
+    this.map.on("draw:edited", function(evt) {
+      evt.layers.eachLayer(function(layer) {
+        
+        // Really there's only one layer to iterate over here, because we have at
+        // most one layer in the editing layer group.
+        generateGeometry(layer);
+      });
+    });
+
+    return this;
+  },
+
+  // ========== Toolbar handlers ==========
+  onClickColorEditModeChange: function(evt) {
+    evt.preventDefault();
+
+    var editMode = $(evt.target).data("edit-mode");
+
+    $(evt.target)
+      .addClass("sp-selected")
+      .siblings()
+      .removeClass("sp-selected");
+    
+    this.colorpickerSettings.editMode = editMode;
+    this.updateColorpicker();
+  },
+
+  onClickColorpicker: function(evt) {
+    evt.preventDefault();
+
+    this.saveWorkingGeometry();
+    this.updateColorpicker();
+    $(".sp-container")
+      .css("left", (evt.pageX - 100) + "px")
```

Comment:
No... good point. It also doesn't work well if the user scrolls the content container after rendering the colorpicker. Resolved.